src/js/components/deployment-view/index.tsx

```diff
@@ -50,20 +52,40 @@ class DeploymentView extends React.Component<Props, void> {
   }
 
   public componentWillMount() {
-    const { loadPreviewAndComments, isUserLoggedIn, match: { params: { entityType, token, id } } } = this.props;
+    const {
+      loadPreviewAndComments,
+      isUserLoggedIn,
+      isLoggingIn,
+      match: {
+        params: {
+          entityType,
+          token,
+          id,
+        },
+      },
+    } = this.props;
 
     if (!isEntityType(entityType)) {
       console.error('Unknown preview type!');
       return;
     }
 
-    loadPreviewAndComments(id, entityType, token, isUserLoggedIn);
+    if (
+      isUserLoggedIn ||
+      (!isUserLoggedIn && !isLoggingIn) // A non-Minard user viewing an open deployment
```

Comment:
What would be the cost of waiting for the user's logged-in status to be resolved at the top level, before rendering any components which require that information?